metaflow/plugins/aip/aip.py

```diff
@@ -889,7 +888,42 @@ def _set_container_labels(self, container_op: ContainerOp):
                 raise ValueError(
                     f"Tag name {annotation_name} must be no more than 63 characters"
                 )
-            container_op.add_pod_annotation(annotation_name, annotation_value)
+            ret[annotation_name] = annotation_value
+
+        # - In context of Zillow CICD self.username == "cicd_compile"
+        # - In the context of a Zillow NB self.username == METAFLOW_USER (user_alias)
+        # - In the context of Metaflow integration tests self.username == USER=$GITLAB_USER_EMAIL
+        owner = self.username
+        if "@" in owner:
+            owner = owner.split("@")[0]
+        ret["zodiac.zillowgroup.net/owner"] = owner
+
+        # Add in Zodiac service and team labels to the aip pods if the environment variable is
+        # present in the notebook (individual profile notebooks only) and set them. These labels
+        # are not being added by poddefaults as they were removed. Workflows launched in project
+        # profiles still get these labels added via poddefaults. Also adds in logging topic
+        # annotation as this value is specific to zodiac service as well.
+        if ZILLOW_ZODIAC_SERVICE and ZILLOW_ZODIAC_TEAM:
+            ret["zodiac.zillowgroup.net/service"] = ZILLOW_ZODIAC_SERVICE
+            ret["zodiac.zillowgroup.net/team"] = ZILLOW_ZODIAC_TEAM
+
+        return ret
+
+    def _set_container_labels(self, container_op: ContainerOp):
+        # TODO(talebz): A Metaflow plugin framework to customize tags, labels, etc.
+        container_op.add_pod_label("aip.zillowgroup.net/aip-pod-default", "true")
+
+        # https://github.com/argoproj/argo-workflows/issues/4525
+        # all argo workflows need istio-injection disabled, else the workflow hangs.
+        container_op.add_pod_label("sidecar.istio.io/inject", "false")
+
+        # add Flow labels as container labels
+        for key, value in self._get_flow_labels().items():
+            container_op.add_pod_label(key, value)
+
+        prefix = "metaflow.org"
+        container_op.add_pod_annotation(f"{prefix}/step", container_op.name)
+        container_op.add_pod_annotation(f"{prefix}/run_id", METAFLOW_RUN_ID)
```

Comment:
Classic question but seems like we have a bit of a mix between:

- Flow vs Container/Step level
- Labels vs Annotations

Is it possible to compile a quick table of what we have as an annotation versus a label and whether they are at a flow or step level? Maybe put it in the PR description so we can summarize what we have?

Reply:
added at the top of the PR